.github/workflows/test.yaml

```diff
@@ -0,0 +1,61 @@
+name: Build and test
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust-version:
+          - stable
+          - beta
+          - nightly
+        cargo-args:
+          - ''
+          - '--all-features'
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust-version }}
+          override: true
+      - name: Run cargo clean
+        uses: actions-rs/cargo@v1
+        with:
+          command: clean
+      - name: Run cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: ${{ matrix.cargo-args }}
+      - name: Run cargo test
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: ${{ matrix.cargo-args }}
+
+      # This is currently failing, even in Travis
+      #- name: Run cargo-tarpaulin
+      #  if: matrix.rust-version == 'stable'
+      #  uses: actions-rs/tarpaulin@v0.1
+      #  with:
+      #    version: 'latest'
+      #    args: '--all-features --exclude-files examples/'
```

Comment:
This would actually need to use `${{ matrix.cargo-args }}`, not sure if `${{ matrix.cargo-args }} --exclude-files examples/` would work as expected.

Reply:
On second thoughts: should this only run on one of the stable runs? e.g.

```
if: matrix.rust-version == 'stable' && matrix.cargo-args != ''
```

Reply:
I think stable is enough.

Reply:
So, running twice? The old Travis configuration had a Matrix of 3 versions (stable, nightly, beta) and 2 arguments, so 6 different configs. This is currently replicating that.

I literally have no clue what's the difference between passing no arguments and passing `--all-features`, and if it makes a difference for tarpaulin.

I'm going to try and see if I can get a stack trace from that error, then it would all be in your hand ;-)

Reply:
> I literally have no clue what's the difference between passing no arguments and passing --all-features, and if it makes a difference for tarpaulin.

Rust allows for some features to be optional. We want to make sure that the compilation/tests pass both with and without those features, but we want to collect the coverage only with all features enabled.
And that's only on stable.

I wouldn't even test beta, and for nightly, I'd only run `cargo test --all-features` just to make sure.

Reply:
Looking at the current time to run tests, it might not be worth complicating the logic.

If I read you answer correctly though, it makes sense to only run tarpaulin for stable **and** `--all-features`, dropping one of the 6 configs.

Adding the stacktraces in the main conversation.